src/main/java/dev/openfga/sdk/api/client/HttpRequestAttempt.java

```diff
@@ -0,0 +1,114 @@
+package dev.openfga.sdk.api.client;
+
+import static dev.openfga.sdk.util.StringUtil.isNullOrWhitespace;
+
+import dev.openfga.sdk.api.configuration.BaseConfiguration;
+import dev.openfga.sdk.errors.*;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+import java.util.concurrent.*;
+
+public class HttpRequestAttempt<T> {
+    private final ApiClient apiClient;
+    private final BaseConfiguration configuration;
+    private final Class<T> clazz;
+    private final String name;
+    private final HttpRequest request;
+
+    public HttpRequestAttempt(
+            HttpRequest request, String name, Class<T> clazz, ApiClient apiClient, BaseConfiguration configuration)
+            throws FgaInvalidParameterException {
+        if (configuration.getMaxRetries() == null) {
+            throw new FgaInvalidParameterException("maxRetries", "Configuration");
+        }
+        this.apiClient = apiClient;
+        this.configuration = configuration;
+        this.name = name;
+        this.request = request;
+        this.clazz = clazz;
```

Comment:
clazz?

Reply:
`class` is a keyword in Java, so it's a common idiom for any variable, field, or method that might be called "class" to be named "clazz"

Examples from the JDK std libraries https://github.com/search?q=repo%3Aopenjdk%2Fjdk%20clazz&type=code and many libraries do the same